### PR TITLE
Posts.com : Milstone : Validation and model specs

### PR DESCRIPTION
**Pull request by ahmed025566**
In this milestone you could see that I have done the following: 

- I have added all the required validations.
- I have created test suite for each model.
Screenshot from 2023-10-28 19-45-38

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  comment = Comment.create(user:, post:, Text: 'Hi Tom!')
+  like = Like.create(user:, post:)
+  it 'should be a vaild post' do
+    expect(post).to be_valid
+  end
+
+  it 'should be not a vaild post if there is no a title' do
+    post.Title = ''
+    expect(post).to_not be_valid
+    post.Title = 'Hello'
```
Comment (tugsem) [inline spec/models/post_spec.rb:15]: - [ ] Please write test statement under this declaration to be sure it's valid with correct values.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  user = User.new(Name: 'Tome', Bio: 'teacher', Photo: 'https://unsplash.com/photos/F_-0BxGuVvo')
+
+  it 'should be valid if it has a name' do
+    expect(user).to be_valid
+  end
+
+  it 'should not be valid if the name is blank' do
+    user.Name = ''
+    expect(user).to_not be_valid
+    user.Name = 'Tom'
```
Comment (tugsem) [inline spec/models/user_spec.rb:13]: - [ ] Please write test statement under this declaration to be sure it's valid with correct values.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  it 'should not be vaild if posts counter recievied none integer value' do
+    user.PostsCounter = 'string'
+    expect(user).to_not be_valid
+    user.PostsCounter = 0
```
Comment (tugsem) [inline spec/models/user_spec.rb:20]: - [ ] Please write test statement under this declaration to be sure it's valid with correct values.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello  @ahmed025566 👋 , 

Good job so far! 👏 You have almost done it!

Alt Text
## Highlights 🚀 
✔️you have added the validations and specs correctly
✔️correct use of gitflow
✔️professional commit messages
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @ahmed025566        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the Awesome work 🥂  Keep Rocking 🥇 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] Professional README file  
- [x] App runs as intended 






 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**